src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultClientNotifierService.cs

```diff
@@ -2,35 +2,46 @@
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Telemetry;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
 // The VSCode OmniSharp client starts the RazorServer before all of its handlers are registered
-// because of this we need to wait until everthing is initialized to make some client requests.
+// because of this we need to wait until everything is initialized to make some client requests.
 // This class takes a TCS which will complete when everything is initialized
 // ensuring that no requests are sent before the client is ready.
 internal class DefaultClientNotifierService : ClientNotifierServiceBase
 {
     private readonly TaskCompletionSource<bool> _initializedCompletionSource;
     private readonly StreamJsonRpc.JsonRpc _jsonRpc;
+    private readonly ITelemetryReporter? _telemetryReporter;
 
-    public DefaultClientNotifierService(StreamJsonRpc.JsonRpc jsonRpc)
+    public DefaultClientNotifierService(StreamJsonRpc.JsonRpc jsonRpc, ITelemetryReporter? telemetryReporter)
     {
         if (jsonRpc is null)
         {
             throw new ArgumentNullException(nameof(jsonRpc));
         }
 
         _jsonRpc = jsonRpc;
+        _telemetryReporter = telemetryReporter;
         _initializedCompletionSource = new TaskCompletionSource<bool>();
     }
 
     public override async Task<TResponse> SendRequestAsync<TParams, TResponse>(string method, TParams @params, CancellationToken cancellationToken)
     {
         await _initializedCompletionSource.Task;
+        using var _ = _telemetryReporter?.StartEventScope(nameof(SendRequestAsync), Severity.Normal, new Dictionary<string, object?>()
+        {
+            {"eventscope.method", method},
+            {"eventscope.params", @params}
```

Comment:
This will almost certainly contain PII so can't be logged like this, and is probably useless if it's logged in hashed form, so I'd remove this (and from the other method that uses it)